src/core/server/http/integration_tests/http_service.test.ts

```diff
@@ -43,10 +43,10 @@ describe('http service', () => {
         router.get({ path: authUrl.auth, validate: false }, async (req, res) =>
           res.ok({ content: 'ok' })
         );
-        // TODO fix me when registerRouter is available before HTTP server is run
-        (root as any).server.http.registerRouter(router);
 
-        await root.setup();
+        const { http } = await root.setup();
+        http.registerRouter(router);
```

Comment:
maybe it seems redundant to call both methods to run a test server. that was done to support such cases without introducing a lot of changes.

I believe we can easily to fix this later. For example, we can extend `KbnTestServer` in a way it creates `root`, declares all routes and after starts a real server.